BatchNormalization.lua

```diff
@@ -154,3 +154,16 @@ function BN:accGradParameters(input, gradOutput, scale)
       self.gradBias:add(scale, self.buffer)
    end
 end
+
+function BN:clearState()
+   for i,v in ipairs{
+      'buffer',
+      'buffer2',
+      'centered',
+      'std',
+      'normalized',
+   } do
+      if self[v] then self[v]:set() end
+   end
```

Comment:
Might be worth factoring that out in to a utils functions, e.g. `nn.utils.clearTensors(self, {...})`